src/runtime-rs/crates/shim-ctl/src/main.rs

```diff
@@ -0,0 +1,37 @@
+use common::{
+	message::Message,
+	types::{ContainerConfig, Request},
+};
+use runtimes::RuntimeHandlerManager;
+use tokio::sync::mpsc::channel;
+
+const MESSAGE_BUFFER_SIZE: usize = 8;
+const WORKER_THREADS: usize = 2;
+
+async fn real_main() {
+	let (sender, _receiver) = channel::<Message>(MESSAGE_BUFFER_SIZE);
+	let manager = RuntimeHandlerManager::new("xxx", sender).await.unwrap();
+
+	let req = Request::CreateContainer(ContainerConfig {
+		container_id: "xxx".to_owned(),
+		bundle: ".".to_owned(),
+		rootfs_mounts: Vec::new(),
```

Comment:
if you set `rootfs_mounts ` to empty, when the runtime is handling rootfs, I think it would fail.

Reply:
That's true, but it's also true of a vast majority of other configuration options since I can't do everything right away.  The current goal is to have just the simplest container started and stopped cleanly, then go back and gradually implement remaining features.  I don't feel comfortable enabling features before I'm ready to work on them.

Reply:
Ah okay, having worked on this a bit more i can see your point now, sorry.  Apparently no container will start with empty `root_mounts` if I understand it correctly now.

Fixing this looks like a significant job though.  If anybody has any hints as to what to put into `rootfs_mounts`, that would be greatly appreciated.

Reply:
My suggestion is to get rid of the `shim-ctl` from this pr this time and let's enable this in the future. What do you think?

Reply:
To be perfectly honest I'd appreciate if we can merge this soon as having it outstanding is an obstacle to my further developing qemu support.  I've been trying to continue development not to waste time but I'm often hitting the same lines of code that we've been polishing here.  When that happens I have to stash my changes, do a rebase & force push to change the PR and finally pop the stash which causes a merge conflict and a manual resolution.  This is tedious, time-consuming and error-prone (apart from the general risk of resulting in bad code there's a particular risk of losing the cosmetic changes again).

So I can break the shim-ctl commit out to a separate PR but I fear that then I'd have to do more weird stuff in my repo to continue development (I'd at least need to duplicate the commit) - or be blocked until both PRs are merged.  I agree that including the commit here is a border case but at the same time, it doesn't seem totally unreasonable so I'd lean towards keeping it here at this point.

Hopefully I'm making some sense. ;-)

Reply:
Yes, that makes sense to me. But I have another concern, `shim-ctl` will be compiled to another binary, is it possible to integrate this into `kata-ctl `? /cc @jodh-intel